cmake/Modules/OpmInterproceduralOptimization.cmake

```diff
@@ -0,0 +1,130 @@
+include(CheckIPOSupported)
+include(TestCXXAcceptsFlag)
+
+# ------------------------------------------------------------------------------
+# opm_interprocedural_optimization
+#
+# Configure interprocedural optimization (IPO) for a target.
+#
+# This function enables IPO (Link Time Optimization) on the specified target.
+# It supports both the standard CMake IPO interface and ThinLTO settings for Clang.
+#
+# The type of optimization applied is controlled via the cache variable:
+#   OPM_INTERPROCEDURAL_OPTIMIZATION_TYPE
+# which can be one of:
+#   - DEFAULT: use standard CMake IPO via INTERPROCEDURAL_OPTIMIZATION_<config>
+#   - ThinLTO: use Clang's ThinLTO with parallelism and cache configuration
+#   - NONE:    disable IPO
+#
+# ThinLTO cache directories are automatically created under:
+#   ${CMAKE_BINARY_DIR}/LTOCache/<config>
+# and registered for clean-up.
+#
+# Supported linkers for ThinLTO customization:
+#   - LLD:       --thinlto-jobs, --thinlto-cache-dir, --thinlto-cache-policy
+#   - GNUgold:   -plugin-opt,jobs=..., cache-dir=..., cache-policy=...
+#   - AppleClang: -mllvm,-threads=..., -cache_path_lto,...
+#
+# Usage:
+#
+#   opm_interprocedural_optimization(
+#     TARGET <target>
+#     [CONFIGURATION_TYPES <config1;config2;...>]
+#   )
+#
+# Arguments:
+#   TARGET               (required) Target to which optimization will be applied.
+#   CONFIGURATION_TYPES  (optional) List of configuration names (e.g. Release).
+#                        If omitted, the default list from:
+#                          OPM_INTERPROCEDURAL_OPTIMIZATION_CONFIGURATION_TYPES
+#                        is used.
+#
+# Cache Variables:
+#   OPM_INTERPROCEDURAL_OPTIMIZATION_TYPE                 (DEFAULT, ThinLTO, NONE)
+#   OPM_INTERPROCEDURAL_OPTIMIZATION_JOBS                 (e.g. 1, 8)
+#   OPM_INTERPROCEDURAL_OPTIMIZATION_CACHE_POLICY         (e.g. "prune_after=604800")
+#   OPM_INTERPROCEDURAL_OPTIMIZATION_CONFIGURATION_TYPES  (e.g. "Release;RelWithDebInfo")
+#
+# ------------------------------------------------------------------------------
+
+
+check_ipo_supported (RESULT ipo_supported)
+
+# Only test ThinLTO if using Clang
+if(ipo_supported AND (CMAKE_CXX_COMPILER_ID MATCHES "Clang"))
+  check_cxx_accepts_flag ("-flto=thin" has_thin_lto_support)
+endif()
+
+# define default values for LTO
+if(ipo_supported AND has_thin_lto_support)
+  set(opm_ipo_type ThinLTO)
+elseif (ipo_supported)
+  set(opm_ipo_type DEFAULT)
+else()
+  set(opm_ipo_type NONE)
+endif()
+
+set(OPM_INTERPROCEDURAL_OPTIMIZATION_TYPE ${opm_ipo_type} CACHE STRING "Default OPM interprocedural optimization type. Possible values: DEFAULT, ThinLTO, NONE")
```

Comment:
Please add
```cmake
set_property(CACHE OPM_INTERPROCEDURAL_OPTIMIZATION_TYPE PROPERTY STRINGS DEFAULT ThinLTO NONE)
```
this way gui frontends get a dropdown list with available alternatives. Also please consider 'ON ThinLTO OFF'